### PR TITLE
Allow to specify global user options in the operator configuration

### PR DESCRIPTION
**Pull request by alexeyklyukin**
Introduce new api_roles_configuration parameter, which is a map of name, value configuration parameters that are applied to the role obtained from the API with ALTER ROLE SET.

As a byproduct, add support for parameters to the PgUser structure, making it easier to implement support for custom options in other places, i.e. in the `users` field  of the cluster manifest.

There is a special case for the search_path in order to strip quotes from the value that would be normally quoted in order to be put into the configMap value when multiple search paths are specified.

Added the parameter configuration description in the README, we may find a better place to put those in the future.

### CODE REVIEW

File: manifests/configmap.yaml [hunk 1]
```diff
@@ -39,3 +39,4 @@ data:
   pod_terminate_grace_period: 5m
   pdb_name_format: "postgres-{cluster}-pdb"
   eol_node_label: "eol:true"
+  api_roles_configuration: "log_statement:all"
```
Comment (Jan-M) [inline manifests/configmap.yaml]: is there a reason you are not calling it teams_api_roles_configuration?

Comment (alexeyklyukin) [reply manifests/configmap.yaml]: no particular one, good idea.

**Comment by alexeyklyukin** [comment]
👍

File: pkg/util/config/util.go [hunk 2]
```diff
@@ -166,6 +169,55 @@ func processField(value string, field reflect.Value) error {
...
+type parserState int
+
+const (
+	Plain parserState = iota
+	DoubleQuoted
+	SingleQuoted
+)
+
+// Split the pair candidates by commas not located inside open quotes
+// Escape characters are not supported for simplicity, as we don't
```
Comment (zerg-junior) [inline pkg/util/config/util.go]: Please add an explicit check for escape characters.

File: manifests/configmap.yaml [hunk 3]
```diff
@@ -39,3 +39,4 @@ data:
   pod_terminate_grace_period: 5m
   pdb_name_format: "postgres-{cluster}-pdb"
   eol_node_label: "eol:true"
+  team_api_role_configuration: "log_statement:all"
```
Comment (zerg-junior) [inline manifests/configmap.yaml]: Please add  an example of specifying multiple config params. I feel the YAML comment will suffice.

Comment (alexeyklyukin) [reply manifests/configmap.yaml]: I'm reluctant to add anything to this configmap just for the demonstration purposes, as it has been serving a good place to show the operator defaults before and, obviously, I don't want to modify the operator defaults for this particular purpose. Instead, I've added the string with multiple keys to the parameter description in the README.

**Comment by alexeyklyukin** [comment]
👍

File: manifests/configmap.yaml (hunk 3 above)
Comment (zerg-junior) [reply manifests/configmap.yaml]: i think that's OK

**Comment by zerg-junior** [comment]
:+1:

**Pull request merged**